backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -110,9 +108,15 @@ public SubjectComment commentAdd(
         @PathVariable Long subjectId,
         @RequestBody AddCommentRequest addCommentRequest
     ){
+        SubjectHubUser requester = getRequestingUser();
+        if(requester == null){
+            Application.log.error("Comment posted by null user, assuming test and posting as userId: 1");
+            requester = subjectHubUserRepository.findOne(Long.parseLong("1"));
+            //not sure how else to implement this such that a user can be specified during tests...
```

Comment:
You should just throw an exception inside getRequestingUser is null and not do the null check here at all. It's okay to just comment out the test for now and I'll fix it later.

Reply:
Thanks! implemented and pushed. trying to learn how to merge now. However, because of the exception handling problems with my branch, my branch should fail and master should work. (copied code, 2 lines that arent supported on branch, from master)